app/com/linkedin/drelephant/exceptions/AnalyzedLog.java

```diff
@@ -0,0 +1,172 @@
+                        m = match("job_[0-9]+_[0-9]+");
+                        while(m.find()){
+                                allMatches.add(m.group());
+                        }
+                        if(match("ERROR - Job run failed!").find()){
+                                this.state = "SCRIPT FAILED";
+                                this.exception.setType("script");
+                                this.list = allMatches;
+                                this.exception.setExceptionChain(matchExceptionChain(".+\\n(?:.+\\tat.+\\n)+(?:.+Caused by.+\\n(?:.*\\n)?(?:.+\\s+at.+\\n)*)*"));
+                        }
+
+                        else {
+                                this.state ="Azkaban Fail";
+                                this.exception.setType("azkaban");
+                                m = match("\\d{2}[-/]\\d{2}[-/]\\d{4} \\d{2}:\\d{2}:\\d{2} PDT [^\\s]+ (?:ERROR|WARN|FATAL|Exception) .*\\n");
+                                if (m.find()) {
+                                        this.exception.addEventException(stringToExceptionEvent(m.group()));
+                                }
+                        }
+                }
+
+                else if(match("Finishing job [^\\s]+ attempt: [0-9]+ at [0-9]+ with status KILLED").find()){            // Killed Azkaban Job log
+                        m = match("job_[0-9]+_[0-9]+");
+                        while(m.find()){
+                                allMatches.add(m.group());
+                        }
+                        this.state = "KILLED";
+                        this.list= allMatches;
+                        //**Incomplete**
+
+                }
+
+
+                else if (match("Job failed as tasks failed").find()){                                                   // Failed MR Job log
+                        this.state = "FAILED";
+                        m = match("Task failed (task_[0-9]+_[0-9]+_[mr]_[0-9]+)");
+                        while(m.find()){
+                                allMatches.add(m.group(1));
+                        }
+                        this.list = allMatches;
+                        this.exception.setExceptionChain(matchExceptionChain(".*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*"));
+                }
+
+                else if (match("Job failed as tasks killed").find()){                                                   // Killed MR Job log
+                        this.state = "KILLED";
+                        m = match("Task killed (task_[0-9]+_[0-9]+_[mr]_[0-9]+)");
+                        while(m.find()){
+                                allMatches.add(m.group(1));
+                        }
+                        this.list = allMatches;
+                        this.exception.setExceptionChain(matchExceptionChain(".*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*"));
```

Comment:
Better to create an ExceptionLoggingEvent object at the top, set Exception chain for the object here and finally set ```this.exception = <object>``` similar to how you handled allMatches.